#
# old_revision [cc8513f94aa94550d600cd8a5d42475dbf7249c0]
#
# patch "libpurple/Makefile.am"
#  from [2112ccec7c2eda63afa57cf4f0c7bbff8347dc42]
#    to [aab882736bf9035cd4f5394c48a171b8f2b1475e]
# 
# patch "libpurple/protocols/jabber/auth.c"
#  from [12caf62ded6c3314a811a59f8730fc801420cd8c]
#    to [1f4804cf7f182a201a16343a14fe6c56b9a476d5]
# 
# patch "libpurple/protocols/jabber/libxmpp.c"
#  from [e90c78b600e55ee816eae95ffc4c0fd69fdd785b]
#    to [589db9e206bb677d6252063048d96aafa79e7347]
# 
# patch "libpurple/protocols/myspace/myspace.h"
#  from [3ff7642b4fa1325f420abb169ac45be9d51bdef5]
#    to [e2dde079de3df378e3bdb2aa3e91921788d7862f]
#
============================================================
--- libpurple/Makefile.am	2112ccec7c2eda63afa57cf4f0c7bbff8347dc42
+++ libpurple/Makefile.am	aab882736bf9035cd4f5394c48a171b8f2b1475e
@@ -237,6 +237,8 @@ libpurple_la_LIBADD = \
 	$(LIBXML_LIBS) \
 	$(LIBNM_LIBS) \
 	$(INTLLIBS) \
+	$(SASL_LIBS) \
+	$(MEANWHILE_LIBS) \
 	-lm
 
 AM_CPPFLAGS = \
============================================================
--- libpurple/protocols/jabber/auth.c	12caf62ded6c3314a811a59f8730fc801420cd8c
+++ libpurple/protocols/jabber/auth.c	1f4804cf7f182a201a16343a14fe6c56b9a476d5
@@ -452,9 +452,11 @@ jabber_auth_start(JabberStream *js, xmln
 void
 jabber_auth_start(JabberStream *js, xmlnode *packet)
 {
-#ifndef HAVE_CYRUS_SASL
+	/* Adium changes: If the avoid_sasl_for_plain_and_digest_md5_auth pref is TRUE, use built-in digest-md5 and plain if possible.
+	 * This is a workaround since PLAIN is broken on Mac OS X 10.4's sasl installation; digest-md5 is retained to minimize the delta versus
+	 * im.pidgin.pidgin.
+	 */
 	gboolean digest_md5 = FALSE, plain=FALSE;
-#endif
 
 	xmlnode *mechs, *mechnode;
 
@@ -473,47 +475,57 @@ jabber_auth_start(JabberStream *js, xmln
 		return;
 	}
 
-#ifdef HAVE_CYRUS_SASL
 	js->sasl_mechs = g_string_new("");
-#endif
 
 	for(mechnode = xmlnode_get_child(mechs, "mechanism"); mechnode;
 			mechnode = xmlnode_get_next_twin(mechnode))
 	{
 		char *mech_name = xmlnode_get_data(mechnode);
-#ifdef HAVE_CYRUS_SASL
+
 		g_string_append(js->sasl_mechs, mech_name);
 		g_string_append_c(js->sasl_mechs, ' ');
-#else
+
 		if(mech_name && !strcmp(mech_name, "DIGEST-MD5"))
 			digest_md5 = TRUE;
 		else if(mech_name && !strcmp(mech_name, "PLAIN"))
 			plain = TRUE;
-#endif
 		g_free(mech_name);
 	}
 
 #ifdef HAVE_CYRUS_SASL
+if (!purple_prefs_get_bool("/plugins/prpl/jabber/avoid_sasl_for_plain_and_digest_md5_auth"))
+{
+	purple_debug_info("jabber", "using sasl");
+
 	js->auth_type = JABBER_AUTH_CYRUS;
 
 	jabber_sasl_build_callbacks(js);
 
 	jabber_auth_start_cyrus(js);
-#else
+}
+else
+#endif
+{
+	purple_debug_info("jabber", "avoiding sasl for plain/digest md5");
 
 	if(digest_md5) {
 		xmlnode *auth;
 
 		js->auth_type = JABBER_AUTH_DIGEST_MD5;
+#ifdef HAVE_CYRUS_SASL
+		js->sasl_state = SASL_OK;
+#endif
 		auth = xmlnode_new("auth");
 		xmlnode_set_namespace(auth, "urn:ietf:params:xml:ns:xmpp-sasl");
 		xmlnode_set_attrib(auth, "mechanism", "DIGEST-MD5");
-
+		
 		jabber_send(js, auth);
 		xmlnode_free(auth);
 	} else if(plain) {
 		js->auth_type = JABBER_AUTH_PLAIN;
-
+#ifdef HAVE_CYRUS_SASL
+		js->sasl_state = SASL_OK;
+#endif
 		if(js->gsc == NULL && !purple_account_get_bool(js->gc->account, "auth_plain_in_clear", FALSE)) {
 			char *msg = g_strdup_printf(_("%s requires plaintext authentication over an unencrypted connection.  Allow this and continue authentication?"),
 					js->gc->account->username);
@@ -527,14 +539,27 @@ jabber_auth_start(JabberStream *js, xmln
 			g_free(msg);
 			return;
 		}
+
 		finish_plaintext_authentication(js);
 	} else {
+#ifdef HAVE_CYRUS_SASL
+		/* digest_md5 and plain are not options; proceed to attempt cyrus-sasl */
+		purple_debug_info("jabber", "using sasl even though we were avoiding it because no DIGEST_MD5 or PLAIN for this server");
+
+		js->auth_type = JABBER_AUTH_CYRUS;
+	
+		jabber_sasl_build_callbacks(js);
+	
+		jabber_auth_start_cyrus(js);
+#else
 		purple_connection_error_reason (js->gc,
-				PURPLE_CONNECTION_ERROR_AUTHENTICATION_IMPOSSIBLE,
-				_("Server does not use any supported authentication method"));
+										PURPLE_CONNECTION_ERROR_AUTHENTICATION_IMPOSSIBLE,
+										_("Server does not use any supported authentication method"));
+		
+#endif
 	}
-#endif
 }
+}
 
 static void auth_old_result_cb(JabberStream *js, xmlnode *packet, gpointer data)
 {
============================================================
--- libpurple/protocols/jabber/libxmpp.c	e90c78b600e55ee816eae95ffc4c0fd69fdd785b
+++ libpurple/protocols/jabber/libxmpp.c	589db9e206bb677d6252063048d96aafa79e7347
@@ -256,6 +256,10 @@ init_plugin(PurplePlugin *plugin)
 	if ((ret = sasl_client_init(NULL)) != SASL_OK) {
 		purple_debug_error("xmpp", "Error (%d) initializing SASL.\n", ret);
 	}
+	
+	/* Adium hack for Mac OS X 10.4 support */
+	purple_prefs_add_none("/plugins/prpl/jabber");
+	purple_prefs_add_bool("/plugins/prpl/jabber/avoid_sasl_for_plain_and_digest_md5_auth", FALSE);
 #endif
 	jabber_register_commands();
 	
============================================================
--- libpurple/protocols/myspace/myspace.h	3ff7642b4fa1325f420abb169ac45be9d51bdef5
+++ libpurple/protocols/myspace/myspace.h	e2dde079de3df378e3bdb2aa3e91921788d7862f
@@ -41,7 +41,7 @@
 #include "notify.h"
 #include "plugin.h"
 #include "accountopt.h"
-#include "version.h"
+#include "libpurple/version.h"
 #include "cipher.h"     /* for SHA-1 */
 #include "util.h"       /* for base64 */
 #include "debug.h"      /* for purple_debug_info */
